### ISSUE TITLE
Image compression before heading to S3

### ISSUE THREAD

**Issue by Lunteer** [humanmade/S3-Uploads#277]
We would like to compress images before they are sent to S3. Is there a way to hook into S3-Uploads to do this? We would like to find a server side solution (instead of a plugin) any suggestions? If nothing server side any plugin suggestions or ones you know work with S3-uploads.

Thanks in advance!

**Comment by uglyrobot** [comment]
Use the power of the cloud. You could setup a serverless function like this to automatically compress any images that get uploaded to s3: https://github.com/sourcey/s3-image-optimizer

**Comment by Lunteer** [comment]
@uglyrobot Thanks so much for the suggestion!

Trying to understand and think I got it but wanted to double check, I need to setup S3-image-optimizer **and** setup on a server https://github.com/imagemin/imagemin or https://github.com/sourcey/s3-image-optimizer includes imagemin?

Thanks again for the suggestion and help!

**Comment by uglyrobot** [comment]
That's not my software so I can't really support it, just use the docs there to set it up. It's a serverless lambda function you can run on your aws account that watches a given bucket for new images, then compress them.

**Comment by Lunteer** [comment]
Closing this issue as we found a much better way to handle compression and image optimization. We are basically not touching the original image.

[Serverless Image Handler](https://aws.amazon.com/solutions/serverless-image-handler/) [source code](https://github.com/awslabs/serverless-image-handler).

We implemented this solution and use srcset to server various image sizes for speed optimization. The best thing is the original image is not touched, which is better for us as we service clients who might not want their images touched.
